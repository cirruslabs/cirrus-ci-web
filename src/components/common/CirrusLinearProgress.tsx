import React from 'react';

import LinearProgress from '@mui/material/LinearProgress';
import { makeStyles } from '@mui/styles';

const useStyles = makeStyles(theme => {
  return {
    progress: {
      backgroundColor: theme.palette.success.main,
    },
  };
});

const CirrusLinearProgress = () => {
  let classes = useStyles();
  return (
    <LinearProgress
      variant="indeterminate"
      classes={{
        barColorPrimary: classes.progress,
      }}
    />
  );
};

export default CirrusLinearProgress;
