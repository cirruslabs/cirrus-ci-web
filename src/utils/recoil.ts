// https://recoiljs.org/docs/guides/atom-effects/#local-storage-persistence
import { DefaultValue } from 'recoil';

export const localStorageEffect =
  key =>
  ({ setSelf, onSet }) => {
    const savedValue = localStorage.getItem(key);
    if (savedValue != null) {
      setSelf(JSON.parse(savedValue));
    }

    onSet(newValue => {
      if (newValue instanceof DefaultValue) {
        localStorage.removeItem(key);
      } else {
        localStorage.setItem(key, JSON.stringify(newValue));
      }
    });
  };
