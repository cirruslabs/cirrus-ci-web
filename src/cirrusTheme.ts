import { atom, selector } from 'recoil';

import { grey, lightGreen, orange, red } from '@mui/material/colors';
import { ThemeOptions } from '@mui/material/styles';

import { localStorageEffect } from 'utils/recoil';

export const prefersDarkModeState = atom({
  key: 'CurrentlyPrefersDarkMode',
  default: false,
  effects_UNSTABLE: [localStorageEffect('CurrentlyPreferredTheme')],
});

export const muiThemeOptions = selector({
  key: 'muiThemeOptions',
  get: ({ get }) => {
    const prefersDarkMode = get(prefersDarkModeState);
    return prefersDarkMode ? muiDarkTheme : muiLightTheme;
  },
});

export const cirrusColorsState = selector({
  key: 'cirrusColorsState',
  get: ({ get }) => {
    const prefersDarkMode = get(prefersDarkModeState);
    return {
      undefined: prefersDarkMode ? grey['600'] : grey['400'],
    };
  },
});

export const cirrusThemeOptions = selector({
  key: 'cirrusThemeOptions',
  get: ({ get }) => {
    const prefersDarkMode = get(prefersDarkModeState);
    return prefersDarkMode ? cirrusDarkTheme : cirrusLightTheme;
  },
});

let cirrusBaseTheme: ThemeOptions = {
  typography: {
    fontFamily: 'Roboto, sans-serif',
  },
  shape: {
    borderRadius: 2,
  },
};

export let cirrusLightTheme: ThemeOptions = {
  ...cirrusBaseTheme,
  palette: {
    mode: 'light',
    primary: {
      main: grey['900'],
      dark: grey['900'],
      light: grey['50'],
      contrastText: grey['50'],
    },
    secondary: {
      main: grey['700'],
      dark: grey['700'],
      light: grey['50'],
      contrastText: grey['50'],
    },
    action: {
      hover: grey['200'],
    },
    success: {
      light: lightGreen['300'],
      main: lightGreen['500'],
      dark: lightGreen['700'],
    },
    error: {
      light: red['300'],
      main: red['500'],
      dark: red['700'],
    },
    warning: {
      light: orange['300'],
      main: orange['500'],
      dark: orange['700'],
    },
  },
  components: {
    MuiChip: {
      styleOverrides: {
        avatar: {
          marginLeft: 0,
          marginRight: 0,
          height: 32,
          width: 32,
        },
      },
    },
    MuiSkeleton: {
      styleOverrides: {
        root: {
          backgroundColor: grey['800'],
        },
      },
    },
    MuiTableCell: {
      styleOverrides: {
        body: {
          padding: 6, // it was changed from v4
        },
      },
    },
  },
};

export let cirrusDarkTheme: ThemeOptions = {
  ...cirrusBaseTheme,
  palette: {
    mode: 'dark',
    primary: {
      main: grey['900'],
      dark: grey['900'],
      light: grey['50'],
      contrastText: grey['50'],
    },
    secondary: {
      main: grey['700'],
      dark: grey['700'],
      light: grey['50'],
      contrastText: grey['50'],
    },
    success: {
      light: lightGreen['400'],
      main: lightGreen['600'],
      dark: lightGreen['800'],
    },
    error: {
      light: red['400'],
      main: red['600'],
      dark: red['800'],
    },
    warning: {
      light: orange['400'],
      main: orange['600'],
      dark: orange['800'],
    },
  },
  components: {
    MuiChip: {
      styleOverrides: {
        avatar: {
          marginLeft: 0,
          marginRight: 0,
          height: 32,
          width: 32,
        },
      },
    },
    MuiTableCell: {
      styleOverrides: {
        body: {
          padding: 6, // it was changed from v4
        },
      },
    },
    MuiTab: {
      styleOverrides: {
        root: {
          '&.Mui-selected': {
            color: grey['50'],
          },
        },
      },
    },
  },
};

export let muiLightTheme: ThemeOptions = {
  palette: cirrusLightTheme.palette,
};

export let muiDarkTheme: ThemeOptions = {
  palette: cirrusDarkTheme.palette,
};

export const cirrusOpenDrawerState = atom({
  key: 'CirrusOpenDrawer',
  default: false,
  effects_UNSTABLE: [localStorageEffect('CirrusOpenDrawer')],
});
