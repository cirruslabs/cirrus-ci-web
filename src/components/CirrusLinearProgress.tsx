import React from 'react';
import LinearProgress from '@material-ui/core/LinearProgress';
import { cirrusColors } from '../cirrusTheme';
import { withStyles } from '@material-ui/core';

const styles = {
  progress: {
    backgroundColor: cirrusColors.success,
  },
};

const CirrusLinearProgress = props => {
  return (
    <LinearProgress
      variant="indeterminate"
      classes={{
        barColorPrimary: props.classes.progress,
      }}
    />
  );
};

export default withStyles(styles)(CirrusLinearProgress);
