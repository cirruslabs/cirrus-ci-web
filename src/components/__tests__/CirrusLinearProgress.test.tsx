import React from 'react';
import renderer from 'react-test-renderer';

import { RecoilRoot } from 'recoil';

import { createTheme } from '@mui/material/styles';
import { ThemeProvider } from '@mui/styles';

import { cirrusLightTheme } from 'cirrusTheme';

import CirrusLinearProgress from 'components/common/CirrusLinearProgress';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <RecoilRoot>
        <ThemeProvider theme={createTheme(cirrusLightTheme)}>
          <CirrusLinearProgress />
        </ThemeProvider>
      </RecoilRoot>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
