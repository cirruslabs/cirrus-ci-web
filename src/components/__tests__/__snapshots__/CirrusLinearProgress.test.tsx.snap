// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
  role="progressbar"
>
  <span
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary CirrusLinearProgress-progress-1 MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
    style={Object {}}
  />
  <span
    className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary CirrusLinearProgress-progress-1 MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
    style={Object {}}
  />
</span>
`;
